src/app/DIAmeterApplication.cpp

```diff
@@ -0,0 +1,1241 @@
+   vector<int> valid_ms1scans, valid_ms2scans;
+   for (int offset=-Params::GetInt("coelution-oneside-scans"); offset<=Params::GetInt("coelution-oneside-scans"); ++offset) {
+       int candidate_ms1scan = ms1_scan_num + offset*scan_gap_;
+       int candidate_ms2scan = ms2_scan_num + offset*scan_gap_;
+       if (candidate_ms1scan < 1 || candidate_ms1scan > max_ms1scan_) { continue; }
+
+       valid_ms1scans.push_back(candidate_ms1scan);
+       valid_ms2scans.push_back(candidate_ms2scan);
+   }
+   // carp(CARP_DETAILED_DEBUG, "^^^^^^^^^^valid_ms1scans:%s \t valid_ms2scans:%s", StringUtils::Join(valid_ms1scans, ',').c_str(), StringUtils::Join(valid_ms2scans, ',').c_str() );
+
+   // Loop through each corresponding ms1scan and ms2scan pair (ppm-based)
+   vector<boost::tuple<double*, double*, int, double*, double*, int>> mz_intensity_arrs_vector;
+   for (pair<vector<int>::const_iterator, vector<int>::const_iterator> f(valid_ms1scans.begin(), valid_ms2scans.begin());
+       f.first != valid_ms1scans.end() && f.second != valid_ms2scans.end(); ++f.first, ++f.second) {
+
+       int curr_ms1scan = *(f.first);
+       int curr_ms2scan = *(f.second);
+
+       int ms1_peak_num = -1; double *ms1_mz_arr = NULL, *ms1_intensity_arr = NULL;
+       int ms2_peak_num = -1; double *ms2_mz_arr = NULL, *ms2_intensity_arr = NULL;
+
+       map<int, boost::tuple<double*, double*, double*, int>>::iterator ms1_intensityIter = ms1scan_mz_intensity_rank_map->find(curr_ms1scan);
+       if (ms1_intensityIter == ms1scan_mz_intensity_rank_map->end()) { carp(CARP_DETAILED_DEBUG, "No intensity found in MS1 scan:%d !!!", curr_ms1scan); }
+       else {
+           ms1_mz_arr = (ms1_intensityIter->second).get<0>();
+           ms1_intensity_arr = (ms1_intensityIter->second).get<1>();
+           ms1_peak_num = (ms1_intensityIter->second).get<3>();
+       }
+
+       map<int, boost::tuple<double*, double*, int>>::iterator ms2_intensityIter = ms2scan_mz_intensity_map->find(curr_ms2scan);
+       if (ms2_intensityIter == ms2scan_mz_intensity_map->end()) { carp(CARP_DETAILED_DEBUG, "No intensity found in MS2 scan:%d !!!", curr_ms2scan); }
+       else {
+              ms2_mz_arr = (ms2_intensityIter->second).get<0>();
+              ms2_intensity_arr = (ms2_intensityIter->second).get<1>();
+              ms2_peak_num = (ms2_intensityIter->second).get<2>();
+       }
+
+       if (ms1_intensity_arr != NULL && ms2_intensity_arr != NULL) {
+           mz_intensity_arrs_vector.push_back(boost::make_tuple(ms1_mz_arr, ms1_intensity_arr, ms1_peak_num, ms2_mz_arr, ms2_intensity_arr, ms2_peak_num));
+       }
+   }
+   map<TideMatchSet::Arr::iterator, boost::tuple<double, double, double>> coelute_map;
+   computePrecFragCoelute(targets, peptides, &mz_intensity_arrs_vector, &coelute_map, charge);
+   computePrecFragCoelute(decoys, peptides, &mz_intensity_arrs_vector, &coelute_map, charge);
+
+   // calculate MS2 p-value
```

Comment:
Maybe a comment that this is different than the p-values you get from the score functions (xcorr p-value / res-ev p-value). Or maybe it is the same?